lib/diagnostic.js

```diff
@@ -8,22 +8,38 @@
 // and returns the minimum from that set
 // assume that all the argument are numbers
 const min = function min () {
+    let result = arguments[0]
 
+    for (let i = 1, maximum = arguments.length; i < maximum; i++) {
+      const current = arguments[i]
+      result = current < result ? current : result
+    }
+    return result
+  }
 }
 
 // Question 2
 // Write a function that takes zero or more arguments
 // and returns the sum of that set
 // assume that all the argument are numbers
 const sum = function sum () {
+    let result = arguments[0]
 
+    for (let i = 1, maximum = arguments.length; i < maximum; i++) {
+      const current = arguments[i]
+      result = current + result[i]
+    }
+    return result
+  }
 }
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
 const newArray = function newArray (/* length, defaultValue */) {
-
+  for (let i = 1; i < newArray.length; i++)
+  Array.prototype.push(defaultValue)
```

Comment:
Make sure you're pushing this to a new array! Right now you are simply calling the push method on it